### PR TITLE
Modernize a bit

### PR DESCRIPTION
**Pull request by simonschmeisser**
this PR slightly modernizes the code, look at the individual commits for more details.

please note that I failed running the unit tests so we might want to wait for #4 before merging

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,9 +33,7 @@ IF(CMAKE_BUILD_TYPE MATCHES [cC][oO][vV][eE][rR][aA][gG][eE])
 ENDIF(CMAKE_BUILD_TYPE MATCHES [cC][oO][vV][eE][rR][aA][gG][eE])
 
 find_package(LibPhoneNumber REQUIRED)
-find_package(Qt5Core)
-find_package(Qt5DBus)
-find_package(Qt5Network)
+find_package(Qt5 REQUIRED COMPONENTS Core DBus Network Test)
```
Comment (minlexx) [inline CMakeLists.txt]: I thnk `Qt5Test` should not be strictly required, it should be optional. Should be required only in `tests` subdirectory (remove `Test` and add `find_package` call in tests dir).

**Comment by minlexx** [comment]
```
fix error handling: ! was applied before comparsion...
fix: control reaches end of non-void function
```
^^ Thise 2 look slightly dangerous, needed to be carefully checked in context, otherwise looks good IMO!

**Comment by Kaffeine** [comment]
The code like `!statement != -1` looks familiar. Projects with own spec extensions prefer to commit generated code, rather than copy svc-by-spec generator.  We had an [issue](https://github.com/TelepathyIM/telepathy-qt/commit/a7941ac1de8ca9f3599d539646638d0c3c82b7d2#diff-4a931512ce65bdc9ca6808adf92d8783) in generator and it appeared in many projects (e.g. [ktp-c-i](https://github.com/KDE/ktp-common-internals/commit/2c75666285464592f5ee8d8d9ea9780e51d9dfb3#diff-fbe68106cdb1940ad4fb962b77b8098c)).

**Comment by Kaffeine** [comment]
I gone ahead and applied the first commit (NULL -> nullptr) to the `staging` branch.

https://github.com/TelepathyIM/telepathy-ofono/pull/5/commits/f5ec853485178bb2290aecf7b8a60c68a1f6444e looks good.

Can you please reword the message of the commit https://github.com/TelepathyIM/telepathy-ofono/pull/5/commits/79cd9e6d8209acc6fa4f587dfa082f8a11f56a31 to fit in 72 chars in the first line? The commit itself is good :). 

See also: https://chris.beams.io/posts/git-commit/#limit-50 (`...consider 72 the hard limit`)

File: sqlitedatabase.cpp [hunk 2]
```diff
@@ -33,6 +33,12 @@ Q_DECLARE_METATYPE(sqlite3*)
 // custom sqlite function "comparePhoneNumbers" used to compare IDs if necessary
 void comparePhoneNumbers(sqlite3_context *context, int argc, sqlite3_value **argv)
 {
+    if (argc < 2) {
```
Comment (Kaffeine) [inline sqlitedatabase.cpp]: It seems that we can be even more strict: `if (argc != 2) ...`

File: CMakeLists.txt (hunk 1 above)
Comment (simonschmeisser) [reply CMakeLists.txt]: Sure, please note however that the test are being build unconditionally at the moment. I moved the `find_package ... test)` nevertheless

**Comment by simonschmeisser** [comment]
> The code like `!statement != -1` looks familiar. Projects with own spec extensions prefer to commit generated code, rather than copy svc-by-spec generator. We had an [issue](https://github.com/TelepathyIM/telepathy-qt/commit/a7941ac1de8ca9f3599d539646638d0c3c82b7d2#diff-4a931512ce65bdc9ca6808adf92d8783) in generator and it appeared in many projects (e.g. [ktp-c-i](https://github.com/KDE/ktp-common-internals/commit/2c75666285464592f5ee8d8d9ea9780e51d9dfb3#diff-fbe68106cdb1940ad4fb962b77b8098c)).

it was actually  `!state == -1` here, which is always false. The left-hand side is converted to bool  and negated and only after that compared to `-1` (bool is promoted to integer so its `1 == -1` or `0 == -1`).  Your statement would be always true (same argument).  :)

**Comment by simonschmeisser** [comment]
I fixed the issues you two found, thanks for the quick review!

**Comment by simonschmeisser** [comment]
tests still pass so this could be merged: https://travis-ci.com/simonschmeisser/telepathy-ofono/builds/121872266

**Pull request merged**